lib/ecto/changeset.ex

```diff
@@ -729,6 +729,29 @@ defmodule Ecto.Changeset do
   end
 
   @doc """
+  Validates a change, of type enum, is a subset of the given enumerable. Like
+  validate_inclusion for lists.
+
+  ## Options
+
+    * `:message` - the message on failure, defaults to "\#{x} is invalid"
+
+  ## Examples
+
+      validate_subset(changeset, :pets, ["cat", "dog", "parrot"])
+      validate_subset(changeset, :lottery_numbers, 0..99)
+
+  """
+  @spec validate_subset(t, atom, Enum.t) :: t
+  def validate_subset(changeset, field, data, opts \\ []) do
+    validate_change changeset, field, {:subset, data}, fn _, values ->
+      Enum.flat_map values, fn(x) ->
```

Comment:
Good point about internationalization. Although it would be nice to know what value caused the error, a problem that any array validation would suffer from. Is Ecto.Changeset:46 doing some sort of internationalize work `    less_than:                {&</2,  "must be less than %{count}"}`?

Is it possible to adopt that syntax for multiple errors per field, or is the following preferred to generate a single validation error, free of internationalization problems?

```
case Enum.any?(value, fn(x) -> not x in data end) do
   true -> [{field, message(opts, "has an invalid entry")}]
   false -> []
end
```